### PR TITLE
Support MHD in the GAMER frontend

### PR DESCRIPTION
**Pull request by hyschive**
This PR adds MHD support for the GAMER frontend.

Should I add tests for this new feature? If so, could anyone remind me how to upload a new dataset to the yt server? The existing GAMER datasets do not incorporate magnetic field.

- [x] Code passes flake8 checker
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by jzuhone** [comment]
@hyschive I can upload a dataset if you like. Can I generate one using the new MHD test problems in gamer? Then you can add answer tests.

**Comment by hyschive** [comment]
> @hyschive I can upload a dataset if you like. Can I generate one using the new MHD test problems in gamer? Then you can add answer tests.

@jzuhone That would be great! Perhaps we can use the Orszag-Tang vortex test (`src/TestProblem/Hydro/MHD_OrszagTangVortex`). It will generate 21 outputs. For your reference, an example image of `Data_000017` using the script `plot_slice.py` can be downloaded using `curl -JO http://use.yt/upload/388322d7`

File: yt/frontends/gamer/data_structures.py [hunk 1]
```diff
@@ -246,16 +246,20 @@ def _set_code_unit_attributes(self):
             setdefaultattr( self, 'mass_unit',   self.quan(self.parameters['Unit_M'], 'g' ) )
             setdefaultattr( self, 'time_unit',   self.quan(self.parameters['Unit_T'], 's' ) )
 
+            if self.mhd:
```
Comment (matthewturk) [inline yt/frontends/gamer/data_structures.py]: Can we get this somehow from `parameters` instead of `self.mhd`?

Comment (hyschive) [reply yt/frontends/gamer/data_structures.py]: It is just an alias of `self. parameters["Magnetohydrodynamics"]`. See https://github.com/yt-project/yt/pull/2306/files#diff-0b29b3dd04114f5c2572cf5015f9a505R323

**Comment by jzuhone** [comment]
This looks great--we need to get the new dataset up on the build box. Does @Xarthisius still handle that?

**Comment by Xarthisius** [comment]
> This looks great--we need to get the new dataset up on the build box. Does @Xarthisius still handle that?

Yeah, file uploaded.

**Comment by jzuhone** [comment]
@hyschive can you bump the answer number for the GAMER tests here to version 5?

https://github.com/yt-project/yt/blob/cab1f48f982c28fbd1f8c1075ce3f1a0db5e686c/tests/tests.yaml#L29-L31

**Comment by jzuhone** [comment]
If the tests pass I will merge.

**Comment by jzuhone** [comment]
The appveyor test failure is unrelated as far as I can tell--several PRs have failed because of something related to the `numpy.clip` function in NumPy 1.17. I'm going to go ahead and merge.

**Comment by jzuhone** [comment]
This is it: 

https://github.com/numpy/numpy/blob/master/doc/release/1.17.0-notes.rst#clip-now-uses-a-ufunc-under-the-hood

the Appveyor box must be using NumPy 1.17 but the other tests aren't.

**Pull request merged**